pkg/snapshot/velero.go

```diff
@@ -379,6 +393,51 @@ func getVersion(nameSpace string) (string, error) {
 	return serverStatus.Status.ServerVersion, nil
 }
 
+func getVeleroPod(ctx context.Context, clientset *kubernetes.Clientset, namespace string) (string, error) {
+	veleroLabels := map[string]string{
+		"component": "velero",
+		"deploy":    "velero",
+	}
+
+	veleroPods, err := clientset.CoreV1().Pods(namespace).List(context.TODO(), metav1.ListOptions{
+		LabelSelector: labels.SelectorFromSet(veleroLabels).String(),
+	})
+	if err != nil {
+		return "", errors.Wrap(err, "failed to list velero pods before restarting")
+	}
+
+	for _, pod := range veleroPods.Items {
+		if pod.Status.Phase == corev1.PodRunning {
```

Comment:
When pod is running, it's not quite ready until containers are also running. `pod.Status.ContainerStatuses` will have that info.